src/bsp/include/bsp/BSPContainer.h

```diff
@@ -0,0 +1,14 @@
+#ifndef _BSPFACTORY_H
+#define _BSPFACTORY_H
+
+#include "bsp/IBSP.h"
+#include "bsp/IUserInterface.h"
+
+namespace BSPContainer {
+/**
+ * @brief Return an instance of the platform dependent BSP.
+ */
+IBSP* getBSP();
```

Comment:
Just realized something, change for a reference would be more appropriate I think, I know it's rare to return a reference, but it makes it explicit that the Container manages the memory and I think it's legitimate in this singleton case.

Reply:
Good idea